stacks-core/src/codec.rs

```diff
@@ -129,3 +129,190 @@ impl Codec for Script {
 		Ok(Self::from(buffer))
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use std::{io::Cursor, str::FromStr};
+
+	use bdk::bitcoin::{
+		blockdata::{
+			opcodes::all::{OP_CHECKSIG, OP_DUP, OP_EQUALVERIFY, OP_HASH160},
+			script::Builder,
+		},
+		secp256k1::{Message, Secp256k1, SecretKey},
+		Amount, PublicKey,
+	};
+
+	use crate::StacksError;
+
+	use super::*;
+
+	#[test]
+	fn should_serialize_amount() {
+		let amount = Amount::from_sat(10_000);
+		let mut serialized_amount = vec![];
+
+		amount.serialize(&mut serialized_amount).unwrap();
+
+		let expected_amount = hex::decode("0000000000002710").unwrap();
+
+		assert_eq!(serialized_amount, expected_amount);
+	}
+
+	#[test]
+	fn should_deserialize_amount() {
+		let mut serialized_amount =
+			Cursor::new(hex::decode("0000000000002710").unwrap());
+
+		let amount = Amount::deserialize(&mut serialized_amount).unwrap();
+
+		let expected_amount = Amount::from_sat(10_000);
+
+		assert_eq!(amount, expected_amount);
+	}
+
+	#[test]
+	fn should_serialize_recoverable_signature() {
+		let signature = get_recoverable_signature().unwrap();
+		let mut serialized_signature = vec![];
+
+		signature.serialize(&mut serialized_signature).unwrap();
+
+		let expected_serialized_signature = hex::decode("0119874ebfb457c08cedb5ebf01fe13bf4b6ac216b6f4044763ad95a69022bf1ba3cdba26d7ebb695a7144c8de4ba672dddfc602ffa9e62a745d8f7e4206ae6a93").unwrap();
+
+		assert_eq!(serialized_signature, expected_serialized_signature);
+	}
+
+	#[test]
+	fn should_deserialize_recoverable_signature() {
+		let mut serialized_signature = Cursor::new(
+			hex::decode("0119874ebfb457c08cedb5ebf01fe13bf4b6ac216b6f4044763ad95a69022bf1ba3cdba26d7ebb695a7144c8de4ba672dddfc602ffa9e62a745d8f7e4206ae6a93").unwrap()
+		);
+
+		let signature =
+			RecoverableSignature::deserialize(&mut serialized_signature)
+				.unwrap();
+
+		let expected_signature = get_recoverable_signature().unwrap();
+
+		assert_eq!(signature, expected_signature);
+	}
+
+	#[test]
+	fn should_fail_deserialize_recoverable_signature_with_recovery_id_bytes_out_of_bounds(
```

Comment:
The recovery ID can only be 0 - 3 `i32`. A greater number is out of bounds and should fail to deserialize.